Return undefined on cache-miss

illusori opened the pull request:
Not entirely sure this is the correct fix, but on a cache-miss cache-manager-S3 is returning a `null` as the value. cache-manager considers `value !== undefined` to be cacheable, and so happily returns the `null` as the value if you use `wrap`.

Using isCacheableValue fixes this. However it's probably a "more correct" fix if cache-manager-S3 returned `undefined` on a cache-miss. I don't have a PR for that though.

tprobinson commented:
I think I may have figured out the more proper solution:

In `src/index.js`, on line `388`, change:
`waterCb(null, null)`
to:
`waterCb()`

Can you try that out and let me know if it fixes the issue without the `isCacheableValue` change?

illusori commented:
That solution didn't quite work, but it did give me a pointer on what needed fixing, thanks.

I've updated the PR with changes so that we return `undefined` on cache-misses and when finding expired keys, and I've reverted the original change to the README.

Any thoughts?
